packages/sdk/src/internal/messaging/data-service-messaging.ts

```diff
@@ -111,21 +111,43 @@ export class DataServiceMessaging implements Messaging {
     const otherMembersPks = Object.fromEntries(
       otherThreadMembers.map((member) => [member.publicKey.toBase58(), member]),
     );
+    
+    let thisThreadMember = {
+      publicKey: new PublicKey(meMember.publicKey),
+      scopes: fromDataServiceScopes(meMember.scopes),
+      // lastReadMessageTimestamp: new Date(), // TODO: implement
+    };
+    let messages = dialect.messages.map((message) => (
+      message
+    ));
+    let lastMessage = messages[messages.length - 1] || null;
+    let messageText = dialect.messages.map((message) => (
+      serde.deserialize(new Uint8Array(message.text))
+    ));
+    let lastMessageThread = null;
+    if (lastMessage != null) {
+      lastMessageThread = {
+        text: messageText[messageText.length - 1] || "",
+        timestamp: new Date(lastMessage.timestamp),
+        author: lastMessage.owner === this.me.toBase58()
+        ? thisThreadMember
+        : otherMembersPks[lastMessage.owner]!,
+        deduplicationId: lastMessage.deduplicationId ,
+      }
+    };
+    messages[messages.length - 1] || "";
```

Comment:
leftover from testing?

Reply:
Yes leftovers from getting the compiler to stop complaining haha. I'll remove it.